Framework/DataHandling/src/AlignAndFocusPowderSlim.cpp

```diff
@@ -522,6 +533,26 @@ void AlignAndFocusPowderSlim::determinePulseIndices(const API::MatrixWorkspace_s
     }
   }
 
+  // filter bad pulses
+  if (getProperty(PropertyNames::FILTER_BAD_PULSES)) {
+    this->progress(.16, "Filtering bad pulses");
+
+    // get limits from proton_charge
+    const auto [min_pcharge, max_pcharge, mean] =
+        wksp->run().getBadPulseRange(LOG_CHARGE_NAME, getProperty(PropertyNames::FILTER_BAD_PULSES_LOWER_CUTOFF));
+    g_log.information() << "Filtering bad pulses; pcharge outside of " << min_pcharge << " to " << max_pcharge << '\n';
+
+    const auto run_start = wksp->getFirstPulseTime();
+    const auto run_stop = wksp->getLastPulseTime();
+
+    const auto log = dynamic_cast<const TimeSeriesProperty<double> *>(wksp->run().getLogData(LOG_CHARGE_NAME));
+    if (log) {
+      roi = log->makeFilterByValue(min_pcharge, max_pcharge, true, Mantid::Kernel::TimeInterval(run_start, run_stop),
```

Comment:
Should one leave here a code comment reminding the developer that the proton charge log is "special" in that its time points are allocated in the middle of the pulse period, hence argument `centre=true`?